compilation fails in tracer code

I cannot compile ewoms anymore due to compile erros in the tracer code (PR #431). This happens for g++-4.9 with dune-2.4 and g++8.0 with DUNE master.

Will try to fix this myself as I urgently need a working master.

Here are the errors:
```
In file included from /home/mblatt/src/dune/tasklets/ewoms/ebos/eclproblem.hh:63,
                 from /home/mblatt/src/dune/tasklets/ewoms/ebos/ebos.cc:33:
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh: In member function ‘bool Ewoms::EclTracerModel<TypeTag>::linearSolve_(const TracerMatrix&, Ewoms::EclTracerModel<TypeTag>::TracerVector&, Ewoms::EclTracerModel<TypeTag>::TracerVector&)’:
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:384:32: error: ‘LinearSolverScalar’ was not declared in this scope
         Dune::FMatrixPrecision<LinearSolverScalar>::set_singular_limit(1.e-30);
                                ^~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:384:32: note: suggested alternative:
In file included from /home/mblatt/src/dune/tasklets/ewoms/ewoms/common/start.hh:30,
                 from /home/mblatt/src/dune/tasklets/ewoms/ebos/ebos.cc:31:
/home/mblatt/src/dune/tasklets/ewoms/ewoms/linear/parallelbasebackend.hh:79:14: note:   ‘Ewoms::Properties::PTag::LinearSolverScalar’
 NEW_PROP_TAG(LinearSolverScalar);
              ^~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/tasklets/ewoms/ewoms/common/propertysystem.hh:261:12: note: in definition of macro ‘NEW_PROP_TAG’
     struct PTagName; } extern int semicolonHack_
            ^~~~~~~~
In file included from /home/mblatt/src/dune/tasklets/ewoms/ebos/eclproblem.hh:63,
                 from /home/mblatt/src/dune/tasklets/ewoms/ebos/ebos.cc:33:
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:384:50: error: template argument 1 is invalid
         Dune::FMatrixPrecision<LinearSolverScalar>::set_singular_limit(1.e-30);
                                                  ^
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:385:50: error: type/value mismatch at argument 1 in template parameter list for ‘template<class ctype> class Dune::FMatrixPrecision’
         Dune::FMatrixPrecision<LinearSolverScalar>::set_absolute_limit(1.e-30);
                                                  ^
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:385:50: note:   expected a type, got ‘LinearSolverScalar’
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:395:23: error: ‘SeqILU’ in namespace ‘Dune’ does not name a template type
         typedef Dune::SeqILU< TracerMatrix, TracerVector, TracerVector  > TracerPreconditioner;
                       ^~~~~~
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:395:17: note: suggested alternative: ‘SeqILUn’
         typedef Dune::SeqILU< TracerMatrix, TracerVector, TracerVector  > TracerPreconditioner;
                 ^~~~
                 SeqILUn
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:399:9: error: ‘TracerPreconditioner’ was not declared in this scope
         TracerPreconditioner tracerPreconditioner(M, 1);
         ^~~~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:399:9: note: suggested alternative: ‘TracerVector’
         TracerPreconditioner tracerPreconditioner(M, 1);
         ^~~~~~~~~~~~~~~~~~~~
         TracerVector
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:402:30: error: ‘tracerPreconditioner’ was not declared in this scope
                              tracerPreconditioner, tolerance, maxIter,
                              ^~~~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:402:30: note: suggested alternative: ‘tracerConcentration’
                              tracerPreconditioner, tolerance, maxIter,
                              ^~~~~~~~~~~~~~~~~~~~
                              tracerConcentration
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh: In instantiation of ‘bool Ewoms::EclTracerModel<TypeTag>::linearSolve_(const TracerMatrix&, Ewoms::EclTracerModel<TypeTag>::TracerVector&, Ewoms::EclTracerModel<TypeTag>::TracerVector&) [with TypeTag = Ewoms::Properties::TTag::EclProblem; Ewoms::EclTracerModel<TypeTag>::TracerMatrix = Dune::BCRSMatrix<Dune::FieldMatrix<double, 1, 1>, std::allocator<Dune::FieldMatrix<double, 1, 1> > >; typename Ewoms::Properties::GetProperty<TypeTag, Ewoms::Properties::PTag::Scalar>::p::type = double; Ewoms::EclTracerModel<TypeTag>::TracerVector = Dune::BlockVector<Dune::FieldVector<double, 1>, std::allocator<Dune::FieldVector<double, 1> > >]’:
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:288:17:   required from ‘void Ewoms::EclTracerModel<TypeTag>::endTimeStep() [with TypeTag = Ewoms::Properties::TTag::EclProblem]’
/home/mblatt/src/dune/tasklets/ewoms/ebos/eclproblem.hh:779:9:   required from ‘void Ewoms::EclProblem<TypeTag>::endTimeStep() [with TypeTag = Ewoms::Properties::TTag::EclProblem]’
/home/mblatt/src/dune/tasklets/ewoms/ewoms/common/simulator.hh:617:17:   required from ‘void Ewoms::Simulator<TypeTag>::run() [with TypeTag = Ewoms::Properties::TTag::EclProblem]’
/home/mblatt/src/dune/tasklets/ewoms/ewoms/common/start.hh:341:9:   required from ‘int Ewoms::start(int, char**) [with TypeTag = Ewoms::Properties::TTag::EclProblem]’
/home/mblatt/src/dune/tasklets/ewoms/ebos/ebos.cc:44:51:   required from here
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:384:53: warning: unused variable ‘set_singular_limit’ [-Wunused-variable]
         Dune::FMatrixPrecision<LinearSolverScalar>::set_singular_limit(1.e-30);
                                                     ^~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/tasklets/ewoms/ebos/ecltracermodel.hh:385:53: warning: unused variable ‘set_absolute_limit’ [-Wunused-variable]
         Dune::FMatrixPrecision<LinearSolverScalar>::set_absolute_limit(1.e-30);
                                                     ^~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/ebos.dir/build.make:63: CMakeFiles/ebos.dir/ebos/ebos.cc.o] Fehler 1
make[2]: Das Ziel „CMakeFiles/ebos.dir/build“ wurde wegen Fehlern nicht aktualisiert.
make[1]: *** [CMakeFiles/Makefile2:1892: CMakeFiles/ebos.dir/all] Fehler 2
[ 98%] Built target art2dgf
make[1]: Das Ziel „all“ wurde wegen Fehlern nicht aktualisiert.
make: *** [Makefile:141: all] Fehler 2
make: Das Ziel „default_target“ wurde wegen Fehlern nicht aktualisiert.
```

--- a/ebos/ecltracermodel.hh
+++ b/ebos/ecltracermodel.hh
@@ -381,8 +381,8 @@ protected:
     bool linearSolve_(const TracerMatrix& M, TracerVector& x, TracerVector& b)
     {
 #if ! DUNE_VERSION_NEWER(DUNE_COMMON, 2,7)
-        Dune::FMatrixPrecision<LinearSolverScalar>::set_singular_limit(1.e-30);
-        Dune::FMatrixPrecision<LinearSolverScalar>::set_absolute_limit(1.e-30);
+        Dune::FMatrixPrecision<Scalar>::set_singular_limit(1.e-30);
+        Dune::FMatrixPrecision<Scalar>::set_absolute_limit(1.e-30);
 #endif
         x = 0.0;
         Scalar tolerance = 1e-2;

--- a/ebos/ecltracermodel.hh
+++ b/ebos/ecltracermodel.hh
@@ -392,11 +392,15 @@ protected:
         typedef Dune::BiCGSTABSolver<TracerVector> TracerSolver;
         typedef Dune::MatrixAdapter<TracerMatrix, TracerVector , TracerVector > TracerOperator;
         typedef Dune::SeqScalarProduct< TracerVector > TracerScalarProduct ;
+#if DUNE_VERSION_NEWER(DUNE_ISTL, 2,6)
         typedef Dune::SeqILU< TracerMatrix, TracerVector, TracerVector  > TracerPreconditioner;
+#else
+        typedef Dune::SeqILUn< TracerMatrix, TracerVector, TracerVector  > TracerPreconditioner;
+#endif
 
         TracerOperator tracerOperator(M);
         TracerScalarProduct tracerScalarProduct;
-        TracerPreconditioner tracerPreconditioner(M, 1);
+        TracerPreconditioner tracerPreconditioner(M, 0, 1); // results in ILU0
 
         TracerSolver solver (tracerOperator, tracerScalarProduct,
                              tracerPreconditioner, tolerance, maxIter,

